### PR TITLE
Fix v3.6.x (and above) download issues for Mac OSX, Windows

### LINKED ISSUE #33: Cannot download mongodb 3.6.x on macOS
It's trying to download https://downloads.mongodb.org/osx/mongodb-osx-x86_64-3.6.5.tgz

It needs to download https://downloads.mongodb.org/osx/mongodb-osx-ssl-x86_64-3.6.5.tgz

It appears that the non-ssl version of mongodb is no longer available as of 3.6.5 -- is there a reason not to just always use the -ssl version on mac (and maybe other platforms as well)? That would make the update much easier.

### PR DESCRIPTION
**Pull request by cantremember**
Fixes #33 

- 'osx' archive naming changed in v3.5
- 'win32' archive naming changed in v3.5
- not all signatures end with trailing whitespace
- MongoDB site Downloads send folks to 'fastdl.mongodb.org'
- added Test Suite for package/arch/version/os variants
- TODOs for 'linux'; 'legacy', 'static', 'debian92', 'rhel57'
- TypeScript development tooling w/ `npm run build`

## negotiable: Test Suite coverage

to ensure that i produced a solution that *really worked*, i added an extensive data-driven Test Suite. it's great for proving out that the package *works* ... however it's a time sink and its scope of coverage could be reduced

feedback is welcome

## negotiable: TODOs

i noticed the following 'operating system' namespaces when looking at [all the download options](https://www.mongodb.org/dl/linux/) for 'linux':

- 'debian92' and 'rhel57', which are real OS versions for linux that aren't yet flushed out
- 'legacy', which may or may not be related to the "using legacy release" notes in the current code
- 'static', the meaning of which i do not know

i can remove any or all of those TODO comments from this commit, at your discretion

## negotiable: TypeScript tooling

if you'd rather there wasn't an npm script and that the devDependencies didn't bring in `tsc`,
i will happily remove those changes

i could also break them off into a separate PR

## did not run tsc

trying to merge the results of `npm run build` will likely reduce spurious conflicts -- probably not a good thing for the Reviewer

### CODE REVIEW

File: src/mongodb-download.ts [hunk 1]
```diff
@@ -522,6 +540,7 @@ export class MongoDBPlatform {
     } else if (release >= 7.1) {
       name += "71";
     } else {
+      // TODO: 'debian92'
```
Comment (LoneRifle) [inline src/mongodb-download.ts]: I've merged #34 earlier on, so this comment is now inaccurate

**Review by LoneRifle — APPROVED** [review]
lgtm otherwise, thanks for the PR!

**Comment by cantremember** [comment]
you are very welcome, @LoneRifle 👍

**Comment by LoneRifle** [comment]
I've not gotten the chance to test this on my machine, but once I do I'll merge to master and have one additional change to support Ubuntu 18 before kicking it out

**Pull request merged**